url/src/comp.php

```diff
@@ -341,35 +337,88 @@ function replace_url($itemID, $oldURL, $newURL, $shortCode, $TalisGUID, $token){
 	echo_message_to_screen(INFO, "replace\t");
 	fwrite($myfile,"replace" . ",");
 
-	$resource_data = get_item($shortCode, $itemID, $TalisGUID, $token);
-	if($resource_data) {
-		// get the existing web addresses
-		$resource_id = get_resource_id($resource_data);
-		$web_address_array = get_webaddress_array($resource_data);
-		$online_resource = get_online_resource($resource_data);
+	$item_data = get_item($shortCode, $itemID, $TalisGUID, $token);
+	if($item_data) {
+		$resources = get_resources_from_item($item_data);
+		$resource_updated = false;
+		$web_address_found = false;
+		$online_resource_found = false;
+		foreach ($resources as $r) {
+			// get the existing web addresses
+			$resource_id = get_resource_id($r);
+			$web_address_array = get_webaddress_array($r);
+			$online_resource = get_online_resource($r);
+
+			if ($online_resource !== false){
+				$online_resource_found = true;
+			}
 
-		if ($web_address_array) {
-			// add a new web addresses to the existing ones
-			$web_address_array = check_web_addresses($oldURL, $newURL, $web_address_array, "replace");
-			// build the PATCH body
-			$body = build_patch_body($resource_id, $web_address_array, $newURL);
-			echo_message_to_screen(DEBUG, "replace_url patch request body: $body");
-			// if not a dry run - update
-			if ($shouldWritetoLive === "true") {
-				post_url($shortCode, $resource_id, $body, $TalisGUID, $token);
-			} else {
-				fwrite($myfile, "Test Run\r\n");
+			if ($web_address_array) {
+				$web_address_found = true;
+				$resource_updated = true;
+				// add a new web addresses to the existing ones
+				$new_web_address_array = check_web_addresses($oldURL, $newURL, $web_address_array, "replace");
+				// If the online resource url matches the old url then change
+				$new_online_resource = ($online_resource === $oldURL) ? $newURL : $online_resource;
```

Comment:
This is changing some behaviour.  Currently we are changing the online resource URL to be the new URL regardless of what it was previously... this change will only update the online resource if it matches the old url.  Meaning if there are multiple web addresses, and the online resource is set as one which is not being updated, then it will be preserved.

This feels like the right behaviour - but what do others think?

Reply:
At present my colleagues are happy for our new web address to be forced into the online resource button regardless of previous content. This may be the preference just for the current updating process. However, they don't have use cases where they intentionally load multiple web addresses for an item.

Reply:
Now I think about it, maybe it would be better to preserve the existing behaviour for now.  The current patch JSON is presuming there is an online resource, and populating it - so even if there is no online resource set, or perhaps if the online resource is set but to something other than "URI" (eg. DOI / openURL) then it will overwrite this to be a web address link.  

Perhaps in this PR it would be better to keep this behaviour the same, and if that needed to be changed it can be a separate PR which is more properly considering all cases, including if there is no online resource set.

Reply:
We've reverted this change to leave it as current behaviour